server/events.go

```diff
@@ -958,8 +968,10 @@ func (s *Server) remoteServerUpdate(sub *subscription, _ *client, _ *Account, su
 		return
 	}
 	si := ssm.Server
-	node := string(getHash(si.Name))
-	s.nodeToInfo.Store(node, nodeInfo{si.Name, si.Cluster, si.ID, false, si.JetStream})
+	if s.sameDomain(si.Domain) {
```

Comment:
Since there is nothing else after this "if" statement, then could do the same than Derek mentione:
```
if !s.sameDomain(si.Domain) {
    return
}
node := ...
```